### PR TITLE
prowgen: add expose option to prowgen config (DPTP-972)

### PR DESCRIPTION
**Pull request by ray-harris**
https://issues.redhat.com/browse/DPTP-972

### CODE REVIEW

File: pkg/config/load.go [hunk 1]
```diff
@@ -22,6 +22,7 @@ var ProwgenFile = ".config.prowgen"
 // Prowgen holds the information of the prowgen's configuration file.
 type Prowgen struct {
 	Private bool `json:"private,omitempty"`
+	Expose  bool `json:"expose,omitempty"`
```
Comment (stevekuznetsov) [inline pkg/config/load.go]: Could we get some GoDoc for this? What does this feature do?

Comment (ray-harris) [reply pkg/config/load.go]: Added

File: pkg/config/load.go [hunk 2]
```diff
@@ -21,7 +21,15 @@ var ProwgenFile = ".config.prowgen"
 
 // Prowgen holds the information of the prowgen's configuration file.
 type Prowgen struct {
+	// Private indicates that the openshift-ci-operator's github credentials should
```
Comment (stevekuznetsov) [inline pkg/config/load.go]: I think I'd word this slightly differently:

```go
// Private indicates that generated jobs should be marked as hidden 
// from display in deck and that they should mount appropriate git credentials
// to clone the repository under test
```

File: pkg/config/load.go [hunk 3]
```diff
@@ -21,7 +21,15 @@ var ProwgenFile = ".config.prowgen"
...
+	// be made available via kube secrets and that hidden: true should
+	// be added to the job to prevent the prow job's logs from being public.
 	Private bool `json:"private,omitempty"`
+	// Expose indicates that, when the Private field is true, that the github
```
Comment (stevekuznetsov) [inline pkg/config/load.go]:
```go
// Expose declares that jobs should not be hidden from view if they are private.
// This field has no effect if private is not set.
```

File: pkg/config/load.go (hunk 2 above)
Comment (ray-harris) [reply pkg/config/load.go]: Done

File: pkg/config/load.go (hunk 3 above)
Comment (ray-harris) [reply pkg/config/load.go]: Done

**Pull request merged**